index.js

```diff
@@ -76,22 +76,27 @@ program
 
 
 const checkFileExistence = (cb) => {
-    fs.access(config.CONF_FILE, (error) => {
+    return fs.access(config.CONF_FILE, (error) => {
         if (error) {
-            cb(null, null);
-        } else {
-            properties.parse(config.CONF_FILE, {
-                path: true
-            }, cb);
+            return cb(null, {});
         }
+
+        return properties.parse(config.CONF_FILE, {
+            path: true
+        }, (error, parsedConfig) => {
+            if (error) {
+                log(`Error in Parsing ${config.CONF_FILE}: ${error}`);
+            }
+            return cb(null, error ? {} : parsedConfig);
+        });
     });
 };
 
 const mainExecution = (parsedConfig, cb) => {
     parsedConfig = parsedConfig || {};
 
     // allow key to be passed via env
-    if (process.env.LOGDNA_AGENT_KEY || process.env.AGENT_KEY || process.env.INGESTION_KEY) {
+    if (process.env.LOGDNA_AGENT_KEY) {
```

Comment:
if we removed this will this not be backwards compatible?

Reply:
Previously, we didn't have that - added [here](https://github.com/logdna/logdna-agent/commit/4b3bfd4637d4d543b824bdf6e105533b6ffc5037) then now realized it might cause some troubles - and people are used to using `LOGDNA_AGENT_KEY` (even in `k8s`) instead of others.